drivers/w1/Kconfig.zephyr_gpio

```
@@ -0,0 +1,29 @@
+# Configuration options for the Zephyr serial 1-Wire Master driver
+
+# Copyright (c) 2023 Hudson C. Dalpra
+# SPDX-License-Identifier: Apache-2.0
+
+DT_COMPAT_ZEPHYR_GPIO := zephyr,w1-gpio
+
+config W1_ZEPHYR_GPIO
+	bool "1-wire GPIO"
+	default y if $(dt_compat_enabled,$(DT_COMPAT_ZEPHYR_GPIO))
```

No need for `DT_COMPAT_ZEPHYR_GPIO`
```suggestion
	default y
	depends on DT_HAS_ZEPHYR_W1_GPIO_ENABLED
```